qa/abstracts/src/main/java/io/camunda/zeebe/process/test/qa/abstracts/inspections/AbstractProcessDefinitionInspectionsTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2021 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.camunda.zeebe.process.test.qa.abstracts.inspections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.process.test.api.ZeebeTestEngine;
+import io.camunda.zeebe.process.test.inspections.InspectionUtility;
+import io.camunda.zeebe.process.test.qa.abstracts.util.Utilities;
+import io.camunda.zeebe.process.test.qa.abstracts.util.Utilities.ProcessPackNamedElements;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractProcessDefinitionInspectionsTest {
+  @Test
+  void testFindStartEventIdByName() {
+    // given
+    Utilities.deployResource(getClient(), ProcessPackNamedElements.RESOURCE_NAME);
+    Utilities.deployResource(getClient(), ProcessPackNamedElements.RESOURCE_NAME);
+    // when
+    String startEventId =
+        InspectionUtility.findProcessDefinitions()
+            .getBpmnElementId(ProcessPackNamedElements.START_EVENT_NAME);
```

Comment:
❓ Do you think it's good enough UX like this? You probably write tests using ZPT more than I do. I'd imaging it might be inconvenient to have to call `InspectionUtility.findProcessDefinitions()` every time you need an element id. Storing it in a variable doesn't work because the stream will be closed if you use it a 2nd time.

Reply:
I feel like it might be more user friendly if we store the processes in the `ProcessDefinitionsInspections`. Something like this:

```java
  private final Map<String, BpmnModelInstance> processDefinitionsById;

  public ProcessDefinitionInspections(
      final ProcessDefinitionRecordStreamFilter deploymentRecordStreamFilter) {
    processDefinitionsById =
        deploymentRecordStreamFilter.withIntent(ProcessIntent.CREATED).stream()
            .collect(
                Collectors.toMap(
                    record -> record.getValue().getBpmnProcessId(),
                    record ->
                        Bpmn.readModelFromStream(
                            new ByteArrayInputStream(record.getValue().getResource()))));
  }
```

Then we can use it in the test like:

```java
    final ProcessDefinitionInspections processDef = InspectionUtility.findProcessDefinitions();
    final String startEventId =
        processDef.getBpmnElementId(ProcessPackNamedElements.START_EVENT_NAME);
    final String endEventId = processDef.getBpmnElementId(ProcessPackNamedElements.END_EVENT_NAME);
```

To be honest even nicer, IMO, would be if the instead of having the `InspectionUtility.findProcessDefinitions()` return a `ProcessDefinitionInspections` having it return a `ProcessDefinitions` object instead. This could then contain the convenience methods such as finding an element by id, or finding an element by process id and id. Happy to discuss to f2f if it's easier.

Reply:
In fact, I would static import `InspectionUtility` which would allow me to call:
```java
final String startEventId = findProcesses().getBpmnElementId(ProcessPackNamedElements.START_EVENT_NAME);
```
Sorry for the static import stuff. I do not like it for main code as well, but for testing, it is quite handy.

Your proposed solution could potentially lead to unexpected behaviour regarding the order of loading the utility and executing commands with zeebe.

Reply:
To me it feels odd having to call `findProcesses()` every time. I found my processes, why would I do it again? It reads very much like this could be stored in a variable and reused, but that would break.

> Your proposed solution could potentially lead to unexpected behaviour regarding the order of loading the utility and executing commands with zeebe.

Imo this makes more sense. If I find my process definitions and decide to do a deployment after that it makes sense to me that the process definitions I found before don't contain this newly deployed process.

Reply:
The ideal solution would be to stick to the C7 test pattern and make a static call for the method directly:
```java

final String startEventId = getBpmnElementId(ProcessPackNamedElements.START_EVENT_NAME);
```
This would require some class containing this static method which does not fit the rest of the test api. The solution you propose would require a change to the implementation so that the filtering on the record stream is only applied on method call?

Reply:
> The solution you propose would require a change to the implementation so that the filtering on the record stream is only applied on method call?

That was my suggestion indeed.

> The ideal solution would be to stick to the C7 test pattern and make a static call for the method directly

We can totally go with the static method as well. The API as-is is not a holy grail. If it's a better UX to have a static method we should go for it 🤷 
We just have to figure out where to put it. IMO we can fit that in the `inspections` package as well. Maybe something like `ProcessDefinitionInspectionUtility` class or so. We have access to the record stream in a static context, so the should be simple enough.

Reply:
Hey @jonathanlukas! Just tagging you to make sure you saw my previous message 🙂

Reply:
I saw your previous message, sorry I was quite busy on friday. I just pushed something according to your suggestions.